### PR TITLE
Controller vs AbstractController difference

### PR DESCRIPTION
**Pull request by guillbdx**
Updates tip about difference between Controller and AbstractController.

### CODE REVIEW

**Review by HeahDude — APPROVED** [review]
This plays well with #9989. Thanks!

**Review by OskarStark — APPROVED** [review]

File: controller.rst [hunk 1]
```diff
@@ -146,11 +146,13 @@ and many others that you'll learn about next.
...
 .. tip::
 
-    You can extend either ``Controller`` or ``AbstractController``. The difference
-    is that when you extend ``AbstractController``, you can't access services directly
-    via ``$this->get()`` or ``$this->container->get()``. This forces you to write
-    more robust code to access services. But if you *do* need direct access to the
-    container, using ``Controller`` is fine.
+    You can extend either ``Controller`` or ``AbstractController``. The
+    difference is that ``AbstractController`` is more strict and doesn't let you
+    access services via ``$this->get()`` or ``$this->container->get()``.
```
Comment (yceruto) [inline controller.rst]: > and doesn't let you access services via ``$this->get()`` or ``$this->container->get()``.

Except for services defined in `getSubscribedServices()` method.

**Comment by javiereguiluz** [comment]
After reading Yonel's comment (https://github.com/symfony/symfony-docs/pull/9991#discussion_r199480864) I'm starting to think that we should remove this note altogether. The `controller.rst` article is **CRITICAL** for Symfony Docs. Newcomers reading this article could think that the (false) rumors about Symfony being too complex and too academic are true (vs other frameworks which are more pragmatic).

On this introductory article, it could be better to focus on showing only `AbstractController` and injecting the services via the constructor or the actions ... and never mention `Controller` and `$this->get(...)` or `$this->container->...`  Later, in the service locator article, we can explain this in detail. But mentioning all this in the critical and introductory controllers article looks very wrong to me.

**Comment by chalasr** [comment]
I agree with @javiereguiluz, it's more confusing than useful. 
To me, having 2 ways of doing the same thing always makes it harder to understand. 
Using `AbstractController` (i.e. using DI in controllers) is the preferred way to create controllers since 3.4, I'd say it should be the only one to be documented.
(extending `Controller` is the legacy way to me, as well as any `ContainerAware` thing which IMHO we should deprecate in 4.x)

**Comment by guillbdx** [comment]
You're right, this tip is probably confusing for new comers. Thing is that the maker command make:controller, which is recommended in the [Controller introduction article](https://symfony.com/doc/current/controller.html#generating-controllers)  still generates a class extending Controller instead of AbstractController, which is in contradiction with best practices. But that has more to do with the maker command itself than with the documentation.

**Comment by javiereguiluz** [comment]
@guillbdx yes ... we discussed about that in the Maker bundle repository. We didn't want to make the change until 4.1 was released, because AbstractController was missing the getParameter() shortcut. But now that the shortcut was added in 4.1, I think we can safely switch to AbstractController in the maker too.

File: controller.rst (hunk 1 above)
Comment (nicolas-grekas) [reply controller.rst]: Indeed. How can we improve this?

**Comment by javiereguiluz** [comment]
I'm closing this pull request for the reasons given above. Instead:

* Maker has been improved to use `AbstractController` in 4.1 (https://github.com/symfony/maker-bundle/pull/221)
* This `controller.rst` article has been updated to only mention `AbstractController` (https://github.com/symfony/symfony-docs/pull/10017)

Thank you all for the discussion!

**Pull request closed without merging**